app/src/main/java/org/mozilla/fenix/settings/logins/PasswordRevealHelper.kt

```diff
@@ -0,0 +1,42 @@
+package org.mozilla.fenix.settings.logins
+
+import android.text.InputType
+import android.widget.ImageButton
+import android.widget.TextView
+import androidx.appcompat.content.res.AppCompatResources
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.ext.components
+
+fun togglePasswordReveal(
+    passwordText: TextView,
+    revealPasswordButton: ImageButton,
+    password: String? = null
+) {
+    val context = passwordText.context
+
+    // If password null, then it means we're showing in EditLoginFragment then if not null
+    // we're showing in LoginDetailFragment
+    val currText = password ?: passwordText.text
```

Comment:
Do we have to pass in the password here? It's unclear to me why we need this. Shouldn't the passwordText have the password in both cases?

Reply:
Before last commit on EditLoginsFragment and LoginsDetailFragment they were not using same way for refreshing.
EditLogins was using passwordText.text and LoginsDetailFragment was using logins?.password.
So I didn't touch them but looks like with latest commit they are refreshing passwordText with same way.
I'm going to test and update pr if there is no problem.

Reply:
Updated.

Reply:
You can check with video in comment zero (updated video).